GitHubCodespaces.ps1

```diff
@@ -898,6 +1081,316 @@ function Wait-GitHubCodespaceAction
+
+        The Git repo for this module can be found here: http://aka.ms/PowerShellForGitHub
+
+    .PARAMETER OrganizationName
+        Name of the Organization.
+
+    .PARAMETER Visibility
+        Which users can access codespaces in the organization.
+        Disabled means that no users can access codespaces in the organization.
+
+    .PARAMETER UserName
+        The usernames of the organization member(s) who should have access
+        to Codespaces in the organization. Required when visibility is SelectedMembers.
+        The provided list of usernames will replace any existing value.
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .EXAMPLE
+        Set-GitHubCodespaceVisibility -Visibility SelectedMembers -User octocat -Force
+
+    .INPUTS
+        GitHub.User
+
+    .NOTES
+        You must authenticate using an access token with the admin:org scope to use this endpoint.
+
+    .LINK
+        https://docs.github.com/en/rest/codespaces/organizations?apiVersion=2022-11-28#manage-access-control-for-organization-codespaces
+    #>
+    [CmdletBinding(
+        SupportsShouldProcess,
+        ConfirmImpact = 'High')]
+    param (
+        [Parameter(Mandatory)]
+        [string] $OrganizationName,
+
+        [ValidateSet('Disabled', 'SelectedMembers', 'AllMembers', 'AllMembersAndOutsideCollaborators')]
+        [string] $Visibility,
+
+        [Parameter(
+            ValueFromPipeline,
+            ValueFromPipelineByPropertyName)]
+        [string[]] $UserName,
+
+        [switch] $Force,
+
+        [string] $AccessToken
+    )
+
+    Write-InvocationLog
+
+    if (($UserName.Count -gt 0) -and ($Visibility -ne 'SelectedMembers'))
+    {
+        $message = 'You can only specify the UserName parameter when the Visibility is set to ''SelectedMembers'''
+        Write-Log -Message $message -Level Error
+        throw $message
+    }
+
+    $visibilityMap = @{
+        Disabled = 'disabled'
+        SelectedMembers = 'selected_members'
+        AllMembers = 'all_members'
+        AllMembersAndOutsideCollaborators = 'all_members_and_outside_collaborators'
+    }
+    $hashBody = @{ visibility = $visibilityMap[$Visibility] }
+
+    if ($Visibility -eq 'SelectedMembers')
+    {
+        $hashBody.Add('selected_usernames', @($UserName))
+    }
+
+    $params = @{
+        UriFragment = "orgs/$OrganizationName/codespaces/access"
+        Body = ConvertTo-Json -InputObject $hashBody
+        Method = 'Put'
+        Description = 'Set Codespace Visiblity'
+        AccessToken = $AccessToken
+        TelemetryEventName = $MyInvocation.MyCommand.Name
+    }
+
+    if ($Force -and (-not $Confirm))
+    {
+        $ConfirmPreference = 'None'
+    }
+
+    if ($PSCmdLet.ShouldProcess($Visibility, 'Set Codespace Visibility'))
+    {
+        Invoke-GHRestMethod @params
```

Comment:
If our signature isn't expecting to return anything for this method, we should probably be explicit about it in the implementation:

```suggestion
        $null = Invoke-GHRestMethod @params
```